api/v1/mongodbcommunity_types.go

```diff
@@ -339,8 +352,24 @@ func (m MongodConfiguration) GetDBDataDir() string {
 // GetDBPort returns the port that should be used for the mongod process.
 // If port is not specified, the default port of 27017 will be used.
 func (m MongodConfiguration) GetDBPort() int {
-	// When passed as a number "net.port" is unmarshalled into a float64
-	return int(objx.New(m.Object).Get("net.port").Float64(float64(automationconfig.DefaultDBPort)))
+	portValue := objx.New(m.Object).Get("net.port")
+
+	// Underlying map could be manipulated in memory, e.g. via SetDBPort - then it will be as int,
+	// or it could be deserialized from JSON and then integer in an untyped map will be deserialized as float64.
+	// It's behavior of https://pkg.go.dev/encoding/json#Unmarshal that is converting JSON integers as float64.
+	if portValue.IsInt() {
```

Comment:
`portValue` here is a  *objx.Value (https://github.com/stretchr/objx/blob/master/value.go#L10), so it is unfortunately not easily castable

Reply:
Thanks for clarification!

I've just stumbled upon this from the [manual page](https://github.com/stretchr/objx#reading-data) of the `objx`:

```
m := objx.MustFromJSON(`{"name": "Mat", "age": 30}`)

// Get the details
name := m.Get("name").Str()
age := m.Get("age").Int()
```

Since we're using objx this way or another, maybe it's better to use its own unmarshaling and `.Int()` methods?

But I'm leaving the final choice to you.

Reply:
From what I see in the source code it's not doing anything more than regular json.Unmarshall as objx.Map does not have UnmarshallJSON method. They only add tryConvertFloat64() as a postProcess step so you don't get float64 from integer fields. But this will only add fuel to the fire as we expect to get floats in many places.

Reply:
Thanks for the explanation. Understood!